### PR TITLE
created frontend routes

### LINKED ISSUE #9: Create Frontend Routes
## Overview
Let's create the basic route structure for our frontend app based on the [Information Architecture](https://docs.google.com/document/d/1GSS28KIvA8sTAAfJZ-MaOFLwX4_5o7A7njipEdzCV58/edit) that CDS designers made.

### Weeks 1/2 (Fall w7/8)
- Install the `react-router-dom` package to the frontend to get started with routing
- Create frontend routes in `App.jsx`
- Create a placeholder component that will be shown on each unfinished route (e.g. showing the current path of the page - hint: `const location = useLocation();` where `useLocation` is imported from `react-router-dom`)
- Don't worry about `ProtectedRoute`s for now, those will be implemented with auth later

## Acceptance Criteria
- demoing some routes at standup either this week or next week to tech leads and other devs

## Resources
- [CDS Information Architecture](https://docs.google.com/document/d/1GSS28KIvA8sTAAfJZ-MaOFLwX4_5o7A7njipEdzCV58/edit)
- [Example From OC Habitats Frontend](https://github.com/ctc-uci/oc-habitats-frontend/blob/dev/src/App.jsx)

### As always, feel free to pm Claude or Alan on Slack for support

### PR DESCRIPTION
**Pull request by orvielalz**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,26 +1,28 @@
 import React from 'react';
-import logo from './logo.svg';
+// import logo from './logo.svg';
```
Comment (alanchangxyz) [inline src/App.jsx]: let's get rid of the default react logo if we don't need it anymore (the file, and this commented out import)

File: src/App.jsx [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
+import { Route, Switch, useLocation } from 'react-router-dom';
 import './App.css';
```
Comment (alanchangxyz) [inline src/App.jsx:10]: usually would prefer to import stuff in this order: modules, components, styles; also, if `App.css` isn't being used let's not import it (can just comment this out for now)

File: src/App.jsx [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
-      </header>
+    <div>
+      {/* <Header /> */}
+      <Switch key={location.pathname} location={location}>
+        <Route path="/login" render={() => <Home />} />
+        <Route path="/forgot" render={() => <Forgot />} />
+        <Route path="/dogs" render={() => <Dogs />} />
+        <Route path="/dogs/new" render={() => <AddDog />} />
+        <Route path="/facilities" render={() => <Facilities />} />
+        <Route path="*" render={() => ''} />
```
Comment (alanchangxyz) [inline src/App.jsx]: let's add a todo comment before this line to link the 404 page once this is merged

File: src/components/Location.jsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+
+function Location() {
```
Comment (alanchangxyz) [inline src/components/Location.jsx]: arrow function components instead of `function` components pls

```JavaScript
const Location = () => {
    ...
};
```

File: src/App.jsx [hunk 5]
```diff
@@ -1,26 +1,28 @@
...
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div>
+      {/* <Header /> */}
+      <Switch key={location.pathname} location={location}>
+        <Route path="/login" render={() => <Home />} />
+        <Route path="/forgot" render={() => <Forgot />} />
```
Comment (alanchangxyz) [inline src/App.jsx]: i think for the Forgot Password page we can afford to make it `/forgot-password`, `/forgot` isn't exactly fully descriptive of what that page does

File: src/App.jsx [hunk 6]
```diff
@@ -1,26 +1,28 @@
 import React from 'react';
-import logo from './logo.svg';
+// import logo from './logo.svg';
+import { Route, Switch, useLocation } from 'react-router-dom';
 import './App.css';
+import Home from './pages/Home';
+import Forgot from './pages/Forgot';
```
Comment (alanchangxyz) [inline src/App.jsx]: [same as route] we can probably afford to make this `ForgotPassword` instead of `Forgot` to make it more descriptive of what it does

**Review by alanchangxyz — CHANGES_REQUESTED** [review]
just a few changes needed 👍

File: src/App.jsx [hunk 7]
```diff
@@ -1,26 +1,28 @@
...
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div>
```
Comment (claudeisakeeb) [inline src/App.jsx]: Can you please change this dev to a React fragment? That way styling doesn't get messed up because divs have default styling. Basically you can just replace \<div\> with <>

File: src/pages/AddDog.jsx [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
```
Comment (claudeisakeeb) [inline src/pages/AddDog.jsx]: Could you actually please move all of the pages into directories with the same name? ie AddDog.jsx would go into a directory called AddDog, Dogs.jsx would go into a directory called Dogs, etc. This is because these pages will 99.999% end up with css files so we might as well take care of that in this task lol. If you want you can create the CSS files too :)
In the case that you do please use module.css (name the file AddDog.module.css as opposed to AddDog.css, etc.)

**Review by claudeisakeeb — CHANGES_REQUESTED** [review]
Looks great so far good job!

I know these routes were implemented with React Router V5 but could we actually implement this with React Router V6? Among other things V6 offers less memory usage, faster loading times, and better route matching (Read more [here](https://dev.to/narendersaini32/what-s-new-in-react-router-v6-2g8g#:~:text=The%20biggest%20thing%20we%20can,have%20size%20only%202.9kb)).

Basically you'd just have to reinstall `react-router-dom` (You can see how in the article) and then replace Switch with Routes and `render=...` with `component=...`. I think you will find that the result with V6 looks much more elegant. Please let Alan or I know if you need help or get stuck with anything.

**Comment by orvielalz** [comment]
> 
Hello! I tried following the article and I ran into an error:
Compiled with problems:X

ERROR in ./node_modules/react-router-dom/esm/react-router-dom.js

Module build failed (from ./node_modules/react-scripts/node_modules/source-map-loader/dist/cjs.js):
Error: ENOENT: no such file or directory, open 'C:\Users\User\Documents\GitHub\ctc-uci\cell-dogs-frontend\node_modules\react-router-dom\esm\react-router-dom.js'

File: src/App.jsx [hunk 9]
```diff
@@ -1,26 +1,28 @@
...
+// import './App.css';
+import Home from './pages/Home/Home';
+import ForgotPassword from './pages/ForgotPassword/ForgotPassword';
+import Dogs from './pages/Dogs/Dogs';
+import AddDog from './pages/AddDog/AddDog';
+import Facilities from './pages/Facilities/Facilities';
 
-function App() {
+const App = () => {
+  const location = useLocation();
```
Comment (claudeisakeeb) [inline src/App.jsx]: Since we're not using useLocation for anything we can just delete this for now unless I'm missing something?

Comment (claudeisakeeb) [reply src/App.jsx]: You can just make Routes have no props :) ie remove key=... and location=...

File: src/App.jsx [hunk 10]
```diff
@@ -1,26 +1,28 @@
...
+    <>
+      {/* <Header /> */}
+      <Routes key={location.pathname} location={location}>
+        <Route path="/login" element={<Home />} />
+        <Route path="/forgot-password" element={<ForgotPassword />} />
+        {/* TODO: link 404 page when merged */}
+        <Route path="/dogs" element={<Dogs />} />
+        <Route path="/dogs/new" element={<AddDog />} />
+        <Route path="/facilities" element={<Facilities />} />
+        <Route path="*" element="" />
```
Comment (claudeisakeeb) [inline src/App.jsx]: Hmmm since this is here let's move the TODO comment on line 18 to above line 22 and change it so that it's more clear what to do for the 404 team when they have to resolve merge conflicts. You can just say something like TODO: replace element with 404 page.

File: src/App.jsx [hunk 11]
```diff
@@ -1,26 +1,28 @@
...
-        </a>
-      </header>
-    </div>
+    <>
+      {/* <Header /> */}
+      <Routes key={location.pathname} location={location}>
+        <Route path="/login" element={<Home />} />
+        <Route path="/forgot-password" element={<ForgotPassword />} />
+        {/* TODO: link 404 page when merged */}
+        <Route path="/dogs" element={<Dogs />} />
```
Comment (claudeisakeeb) [inline src/App.jsx]: Since the landing page is going to be the doggo page lets just make this path="/"

File: src/components/Location.jsx [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
```
Comment (claudeisakeeb) [inline src/components/Location.jsx:1]: nice

File: src/App.jsx [hunk 13]
```diff
@@ -1,26 +1,28 @@
...
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <>
```
Comment (claudeisakeeb) [inline src/App.jsx]: I think you can safely delete this unnecessary React Fragment (along with its closing tag) because everything is contained in <Routes>

**Review by claudeisakeeb — CHANGES_REQUESTED** [review]
Great job! Just a few minor comments and this should be good.

For future sprints can y'all make your PRs with the /pr command and fill out the autogenerated template? You don't have to do that for this one but as tasks get bigger the template makes it easier for the reviewers :)

**Review by claudeisakeeb — DISMISSED** [review]
lets get this money

**Review by Madhu2244 — APPROVED** [review]
